src/Microsoft.OData.Core/Json/BufferingJsonReader.cs

```diff
@@ -195,14 +182,13 @@ internal bool IsBuffering
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Performance", "CA1825:Avoid zero-length array allocations.", Justification = "<Pending>")]
         public virtual Stream CreateReadStream()
         {
-            IJsonStreamReader streamReader = this.innerReader as IJsonStreamReader;
-            if (!this.isBuffering && streamReader != null)
+            if (!this.isBuffering)
             {
-                return streamReader.CreateReadStream();
+                return this.innerReader.CreateReadStream();
             }
 
-            Stream result = this.Value == null ? Stream.Null :
-                new MemoryStream(Convert.FromBase64String((string)this.Value));
+            Stream result = this.GetValue() == null ? Stream.Null :

```

Comment:
I think it'd be better to call `this.GetValue()` and store the result in a variable instead of calling it twice in the same statement in case the two calls result into reading from the stream twice